pymongo/common.py

```diff
@@ -524,56 +525,78 @@ def validate_tzinfo(dummy, value):
     return value
 
 
-# journal is an alias for j,
-# wtimeoutms is an alias for wtimeout,
-URI_VALIDATORS = {
-    'replicaset': validate_string_or_none,
-    'w': validate_non_negative_int_or_basestring,
-    'wtimeout': validate_non_negative_integer,
-    'wtimeoutms': validate_non_negative_integer,
-    'fsync': validate_boolean_or_string,
-    'j': validate_boolean_or_string,
+# Dictionary where keys are the names of public URI options, and values
+# are lists of aliases for that option. Aliases of option names are assumed
+# to have been deprecated.
+URI_OPTIONS_ALIAS_MAP = {
+    'journal': ['j'],
+    'wtimeoutms': ['wtimeout'],
+    'tls': ['ssl'],
+    'tlsallowinvalidcertificates': ['ssl_cert_reqs'],
```

Comment:
This seems wrong to me. ssl_cert_reqs is a tri-state. It's value can be ssl.CERT_NONE (0), ssl.CERT_OPTIONAL (1), or ssl.CERT_REQUIRED (2). I'm fine with treating CERT_OPTIONAL as the same as CERT_REQUIRED (since CERT_OPTIONAL has no meaning on the client side), but this mapping would seem to make CERT_REQUIRED and tlsallowinvalidcertificates=true as being equivalent when they are actually opposites. tlsallowinvalidcertificates=true is the same as ssl_cert_reqs=ssl.CERT_NONE.